src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1358,7 +1358,9 @@ public void execute() throws GitException, InterruptedException {
                 submoduleInit();
 
                 ArgumentListBuilder args = new ArgumentListBuilder();
-                args.add("submodule", "update");
+
+                // Force the update, in the same manner we force checkouts
+                args.add("submodule", "update", "--checkout", "--force");
```

Comment:
I think this will need to be controlled by an additional flag added to the submodule options.  Based on the large number of Jenkins installations running the git plugin, we are not usually safe to alter an existing behavior.  Instead, we need a flag which enables the new behavior and is selected by the user when the job is defined.